Build with the system nspr library.

Originally based on a Debian patch:
https://salsa.debian.org/chromium-team/chromium/-/blob/master/debian/patches/system/nspr.patch

--- a/base/time/pr_time_unittest.cc
+++ b/base/time/pr_time_unittest.cc
@@ -7,7 +7,7 @@
 
 #include "base/compiler_specific.h"
 #include "base/stl_util.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time/time.h"
 #include "build/build_config.h"
 #include "testing/gtest/include/gtest/gtest.h"
--- a/base/time/time.cc
+++ b/base/time/time.cc
@@ -14,7 +14,7 @@
 #include "base/macros.h"
 #include "base/no_destructor.h"
 #include "base/strings/stringprintf.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time/time_override.h"
 #include "build/build_config.h"
 
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -107,6 +107,9 @@ config("base_flags") {
       "-Wglobal-constructors",
     ]
   }
+  ldflags = [
+    "-lnspr4",
+  ]
 }
 
 config("base_implementation") {
@@ -712,8 +715,6 @@ jumbo_component("base") {
     "third_party/cityhash_v103/src/city_v103.h",
     "third_party/icu/icu_utf.cc",
     "third_party/icu/icu_utf.h",
-    "third_party/nspr/prtime.cc",
-    "third_party/nspr/prtime.h",
     "third_party/superfasthash/superfasthash.c",
     "thread_annotations.h",
     "threading/hang_watcher.cc",
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -339,6 +339,9 @@ def WriteGNNinja(path, platform, host, o
       if not options.no_static_libstdcpp:
         ldflags.append('-static-libstdc++')
 
+      # Additional system libraries that are used.
+      libs.append('-lnspr4')
+
       # This is needed by libc++.
       if not platform.is_mingw():
         libs.append('-ldl')
--- a/third_party/blink/renderer/platform/BUILD.gn
+++ b/third_party/blink/renderer/platform/BUILD.gn
@@ -103,6 +103,9 @@
     "//build/win:default_exe_manifest",
     "//third_party/icu",
   ]
+  ldflags = [
+    "-lnspr4"
+  ]
 }
 
 # This isn't strictly necessary since we can just add the deps to "platform",
--- a/third_party/crashpad/crashpad/handler/BUILD.gn
+++ b/third_party/crashpad/crashpad/handler/BUILD.gn
@@ -140,6 +140,9 @@
       "../third_party/mini_chromium:base",
       "../tools:tool_support",
     ]
+    ldflags = [
+      "-lnspr4"
+    ]
 
     if (crashpad_is_win) {
       if (crashpad_is_in_chromium || crashpad_is_in_dart) {
--- a/chrome/common/search/BUILD.gn
+++ b/chrome/common/search/BUILD.gn
@@ -20,6 +20,9 @@
     "//chrome/common/themes:autogenerated_theme_util",
     "//skia",
   ]
+  ldflags = [
+    "-lnspr4"
+  ]
 }
 
 compiled_action("generate_chrome_colors_info") {
--- a/components/url_formatter/spoof_checks/top_domains/BUILD.gn
+++ b/components/url_formatter/spoof_checks/top_domains/BUILD.gn
@@ -31,6 +31,9 @@
     "//net/tools/huffman_trie:huffman_trie_generator_sources",
     "//url:url",
   ]
+  ldflags = [
+    "-lnspr4",
+  ]
   if (is_ios) {
     libs = [ "UIKit.framework" ]
   }
--- a/components/schema_org/BUILD.gn
+++ b/components/schema_org/BUILD.gn
@@ -33,6 +33,9 @@
     "//base",
     "//url",
   ]
+  ldflags = [
+    "-lnspr4",
+  ]
 }
 
 compiled_action("schema_org_name_data") {
